### PR TITLE
Update sleuth/zipkin dependencies

### LINKED ISSUE #17487: Unable to start microservice application with sleuth/zipkin enabled
##### **Overview of the issue**

When zipkin maven profile is enabled on microservice application, unable to create taskexecutor bean.

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'liquibase' defined in class path resource [io/github/jhipster/sample/config/LiquibaseConfiguration.class]: Unsatisfied dependency expressed through method 'liquibase' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'taskExecutor' defined in class path resource [io/github/jhipster/sample/config/AsyncConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [java.util.concurrent.Executor]: Illegal arguments to factory method 'getAsyncExecutor'; args: ; nested exception is java.lang.IllegalArgumentException: object is not an instance of declaring class
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:800)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:541)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1352)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1195)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:582)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1154)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:908)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:145)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:765)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:445)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:338)
	at io.github.jhipster.sample.JhipsterSampleMicroserviceApp.main(JhipsterSampleMicroserviceApp.java:69)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:49)
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'taskExecutor' defined in class path resource [io/github/jhipster/sample/config/AsyncConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [java.util.concurrent.Executor]: Illegal arguments to factory method 'getAsyncExecutor'; args: ; nested exception is java.lang.IllegalArgumentException: object is not an instance of declaring class
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:658)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:486)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1352)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1195)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:582)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1389)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1309)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:887)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:791)
	... 24 common frames omitted
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [java.util.concurrent.Executor]: Illegal arguments to factory method 'getAsyncExecutor'; args: ; nested exception is java.lang.IllegalArgumentException: object is not an instance of declaring class
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:172)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:653)
	... 38 common frames omitted
Caused by: java.lang.IllegalArgumentException: object is not an instance of declaring class
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154)
	... 39 common frames omitted

##### **Motivation for or Use Case**

Applcation should start when sleuth/zipkin is enabled

##### **Reproduce the error**

- clone https://github.com/jhipster/jhipster-sample-app-microservice
- Force spring-cloud-starter-zipkin version wich is missing to 2.2.8.RELEASE
- run ./mvnw -Pdev,zipkin

##### **JHipster Version(s)**

generator-jhipster@7.5.0

##### **Browsers and Operating System**

- [ ] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by vmartowicz**
fix #17487

This pull request :

- fixes the wrong zipkin dependency
- fixes sleuth async executor
- add a docker compose zipkin

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/pom.xml.ejs [hunk 1]
```diff
@@ -35,7 +35,7 @@
         <repository>
             <id>spring-milestone</id>
             <url>https://repo.spring.io/milestone</url>
-        </repository>
+        </repository>k
```
Comment (DanielFran) [inline generators/server/templates/pom.xml.ejs:38]:
```suggestion
        </repository>
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

**Comment by DanielFran** [comment]
@vmartowicz Can you remove the package-lock.json file?

File: generators/server/templates/src/main/resources/config/application-dev.yml.ejs [hunk 2]
```diff
@@ -255,6 +255,11 @@ spring:
     cache: false
 <%_ if (serviceDiscoveryType || applicationTypeGateway || applicationTypeMicroservice) { _%>
   sleuth:
+    # https://github.com/spring-projects/spring-framework/issues/27844#issuescomment-1002866885
+    # https://github.com/spring-cloud/spring-cloud-sleuth/issues/2094
+    async:
+      enabled: false
+    enabled: true
```
Comment (pascalgrimaud) [inline generators/server/templates/src/main/resources/config/application-dev.yml.ejs:262]: Not sure it should be enabled by default

Comment (vmartowicz) [reply generators/server/templates/src/main/resources/config/application-dev.yml.ejs:262]: I agree we should disable it by default

File: generators/generator-constants.js [hunk 3]
```diff
@@ -77,6 +77,7 @@ const DOCKER_JENKINS = 'jenkins/jenkins:lts-jdk11';
 const DOCKER_SWAGGER_EDITOR = 'swaggerapi/swagger-editor:latest';
 const DOCKER_PROMETHEUS_OPERATOR = 'quay.io/coreos/prometheus-operator:v0.42.1';
 const DOCKER_GRAFANA_WATCHER = 'quay.io/coreos/grafana-watcher:v0.0.8';
+const DOCKER_ZIPKIN = 'openzipkin/zipkin:latest';
```
Comment (DanielFran) [inline generators/generator-constants.js:80]:
```suggestion
const DOCKER_ZIPKIN = 'openzipkin/zipkin:2.23';
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

**Comment by DanielFran** [comment]
@vmartowicz Do not forget to remove the changes of the package-lock.json file

**Comment by DanielFran** [comment]
@vmartowicz Can you update the snapshot files?
https://github.com/jhipster/generator-jhipster/blob/main/package.json#L80

**Comment by vmartowicz** [comment]
@DanielFran When I run the asked command, I got the attached errors. What do I need to do?

[generator-jhipster-17487.txt](https://github.com/jhipster/generator-jhipster/files/7921395/generator-jhipster-17487.txt)
?

**Comment by vmartowicz** [comment]
@DanielFran 
I m always getting the same error. I don't know what to do....

**Comment by mshima** [comment]
Need to run `npm run update-snapshots`.

**Comment by mshima** [comment]
Rebase or merge from upstream before, otherwise the snapshot may be outdated.

**Comment by vmartowicz** [comment]
> Rebase or merge from upstream before, otherwise the snapshot may be outdated.
 yes I did it

> Need to run `npm run update-snapshots`.
Yes but I got the attached errors

I will try again

**Comment by vmartowicz** [comment]
@mshima @DanielFran Hi I succeed to update my snapshots but some frontend tests failed with the following message.

- ngx-infinite-scroll [es2015/esm2015] (https://github.com/orizens/ngx-infinite-scroll.git)
✔ Browser application bundle generation complete.
An unhandled exception occurred: Transform failed with 1 error:
error: Invalid version: "15.2-15.3"
See "/tmp/ng-HryzP3/angular-errors.log" for further details.
##[error]Bash exited with code '127'.
Finishing: TESTS: frontend

I see some other PR with the same tests failed. Is it normal? Or do I need to do something?

**Comment by mshima** [comment]
Transitional dependency failure https://github.com/angular/angular-cli/issues/22606.
Should be fixed soon.

**Comment by vmartowicz** [comment]
@mshima angular/angular-cli#22606. is closed. 
Do I need to do something?

**Comment by mshima** [comment]
@vmartowicz I've rebased and updated the snapshots.

**Comment by vmartowicz** [comment]
@mshima thanks

**Review by mshima — APPROVED** [review]
I don’t know about zipkin.
The PR looks ok.

**Pull request merged**